src/RetargetingHelper/src/Helper.cpp

```diff
@@ -26,112 +26,328 @@ void RetargetingClient::convertYarpVectorPoseIntoTransform(const yarp::sig::Vect
+        m_jointRetargeting.smoother->init(m_jointRetargeting.yarpVector);
 
-    iDynTree::toEigen(m_leftHandTransformYarp).segment<3>(3) =
-        iDynTree::toEigen(m_leftHandTransform.getRotation().asRPY());
+    m_comHeightValue = comHeight;
+    m_comHeightVelocity = 0;
+    m_comConstantHeight = comHeight;
 
-    iDynTree::toEigen(m_rightHandTransformYarp).segment<3>(0) =
-        iDynTree::toEigen(m_rightHandTransform.getPosition());
+    if(m_useCoMHeightRetargeting)
+    {
+        m_comHeight.yarpVector(0) = comHeight;
+        m_comHeight.smoother->init(m_comHeight.yarpVector);
 
-    iDynTree::toEigen(m_rightHandTransformYarp).segment<3>(3) =
-        iDynTree::toEigen(m_rightHandTransform.getRotation().asRPY());
+        // let's read the port to reset the comHeightInput
+        bool okCoMHeight = false;
+        unsigned int attempt = 0;
+        do
+        {
+            if(!okCoMHeight)
+            {
+                auto desiredCoMHeight = m_comHeight.port.read(false);
+                if(desiredCoMHeight != nullptr)
+                {
+                    m_comHeightInputZero = (*desiredCoMHeight)(2);
+                    okCoMHeight = true;
+                }
+            }
+
+            if(okCoMHeight)
+                return true;
+
+            yarp::os::Time::delay(0.001);
+            attempt++;
+        } while (attempt < 100);
+
+        if(!okCoMHeight)
+            yError() << "[RetargetingClient::reset] The CoM height is not coming from the yarp port.";
+        return false;
+    }
 
-    m_leftHandSmoother->init(m_leftHandTransformYarp);
-    m_rightHandSmoother->init(m_rightHandTransformYarp);
+    return true;
 }
 
 void RetargetingClient::getFeedback()
 {
-    if(!m_useHandRetargeting)
-        return;
+    if(m_useHandRetargeting)
+    {
+        auto desiredLeftHandPose = m_leftHand.port.read(false);
+        if(desiredLeftHandPose != nullptr)
+            m_leftHand.yarpVector = *desiredLeftHandPose;
 
-    auto desiredLeftHandPose = m_leftHandTransformPort.read(false);
-    if(desiredLeftHandPose != nullptr)
-        m_leftHandTransformYarp = *desiredLeftHandPose;
+        m_leftHand.smoother->computeNextValues(m_leftHand.yarpVector);
+        convertYarpVectorPoseIntoTransform(m_leftHand.smoother->getPos(), m_leftHandTransform);
 
-    m_leftHandSmoother->computeNextValues(m_leftHandTransformYarp);
-    convertYarpVectorPoseIntoTransform(m_leftHandSmoother->getPos(), m_leftHandTransform);
+        auto desiredRightHandPose = m_rightHand.port.read(false);
+        if(desiredRightHandPose != nullptr)
+            m_rightHand.yarpVector = *desiredRightHandPose;
 
-    auto desiredRightHandPose = m_rightHandTransformPort.read(false);
-    if(desiredRightHandPose != nullptr)
-        m_rightHandTransformYarp = *desiredRightHandPose;
+        m_rightHand.smoother->computeNextValues(m_rightHand.yarpVector);
+        convertYarpVectorPoseIntoTransform(m_rightHand.smoother->getPos(), m_rightHandTransform);
```

Comment:
You could try to write a function that does the same job for both hands and remove code duplications.

Reply:
Associated issue: https://github.com/dic-iit/bipedal-locomotion-controllers/issues/39

Reply:
In https://github.com/robotology/walking-controllers/pull/41/commits/2fdcdfaffe01255cd707c2ffd89a78128bd4615e, I tried to remove the duplicated code using lambda functions.